### PR TITLE
docs(serena): update memories with v0.5.0 implementation learnings

### PR DESCRIPTION
**Pull request by rjmurillo**
## Summary

Updates Serena memories with learnings from the v0.5.0 implementation session where 4 PRs were developed, reviewed, and verified.

## Changes

- **codebase-analysis-observations**: Added 7 HIGH constraints, 4 MED preferences, 5 MED edge cases from multi-agent orchestration
- **roslyn-analyzer-best-practices**: Verified Moq type catalog against actual source, documented phantom vs real symbols, added positive/negative test requirements

## Key Learnings Captured

- Verify Moq types against actual source before adding to MoqKnownSymbols (4 phantom symbols found)
- Every analyzer change needs both positive and negative tests (v0.4.1/v0.4.2 lesson)
- New analyzers require Internet research on the target library API
- `IReturnsResult<TMock>` is the actual return type of `.Returns()`, not `IReturns`
- `It.Ref<T>.IsAny` has a different containing type symbol than `Moq.It`
- Reflection-based completeness tests catch future gaps automatically
- Agent claims about unreachable code must be independently verified

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added multiple new guidance docs including a critical-mistakes prevention guide and a Moq API surface reference.
  * Expanded best-practices with IOperation-specific guidance, lambda/ternary handling, and Moq type-resolution rules.
  * Strengthened analyzer testing requirements (positive/negative/doppelgänger tests, coverage targets) and added SDK/tooling verification guidance.

### CODE REVIEW

File: .serena/memories/roslyn-analyzer-best-practices.md [hunk 1]
```diff
@@ -1,18 +1,61 @@
...
+
+### Test Patterns
+- Use `{|DiagnosticId:code|}` markers for expected diagnostics
+- Test against both Moq 4.8.2 and 4.18.4 via `.WithMoqReferenceAssemblyGroups()`
+- Include doppelganger tests (user-defined classes that resemble Moq types should NOT trigger)
+- Include literal value tests (non-matcher arguments should NOT trigger matcher-specific diagnostics)
+- Use reflection-based completeness tests to guard against gaps (pit of success)
+
+### Coverage Target
+100% block coverage for new analyzer code. Defensive null guards that are provably unreachable should be documented, not tested with impossible scenarios.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .serena/memories/roslyn-analyzer-best-practices.md]: Fixed in 9783013. Added blank lines around all headings and list blocks in roslyn-analyzer-best-practices.md.

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -41,7 +41,7 @@
 
 ---
 
-You are an experienced .NET developer working on Roslyn analyzers for the Moq framework. All code must target **.NET 9** and **C# 13**. Use only official .NET patterns and practices—**never** StackOverflow shortcuts. Keep responses clear, concise, and at a grade 9 reading level. Use plain English, avoid jargon. Follow SOLID, DRY, and YAGNI principles. Respond directly and keep explanations straightforward.
+You are an experienced .NET developer working on Roslyn analyzers for the Moq framework. All code must target **.NET 10** and **C# 14**. Use only official .NET patterns and practices—**never** StackOverflow shortcuts. Keep responses clear, concise, and at a grade 9 reading level. Use plain English, avoid jargon. Follow SOLID, DRY, and YAGNI principles. Respond directly and keep explanations straightforward.
```
[the remark this answers is not in the document]
Comment (rjmurillo) [reply .github/copilot-instructions.md]: Fixed in same commit. Both occurrences reworded.
